src/NServiceBus.Testing/NSB.Testing.Fakes/TestingExtensions.cs

```diff
@@ -0,0 +1,51 @@
+﻿namespace NServiceBus.Testing
+{
+    using System.Linq;
+
+    /// <summary>
+    /// TODO
+    /// </summary>
+    public static class TestingExtensions
+    {
+        /// <summary>
+        /// TODO
+        /// </summary>
+        /// <typeparam name="TMessageType"></typeparam>
+        /// <returns></returns>
+        public static TMessageType FindPublishedMessage<TMessageType>(this TestablePipelineContext context)
```

Comment:
Based on the name my expectation would be that it is possible to **not** find a message and then the result would be null

Reply:
Changed all these to FirstOrDefault and made it clear in the xmldoc that null was returned of none found.